samples/embedded-map/src/components/EmbeddedMap/EmbeddedMap.tsx

```diff
@@ -1,32 +1,38 @@
-import React, { useEffect, useRef } from "react";
+import React, { useEffect, useRef, useState } from "react";
+import clsx from "clsx";
+import { Viewer, TransitionMode } from "mapillary-js";
 import {
     LayoutElement,
     LayoutElementProperties,
 } from "@vertigis/web/components";
-import { Viewer } from "mapillary-js";
+import IconButton from "@vertigis/web/ui/IconButton";
+import Sync from "@vertigis/web/ui/icons/Sync";
+import CenterMap from "@vertigis/web/ui/icons/CenterMap";
+
 // Import the necessary CSS for the Mapillary viewer to be styled correctly.
 import "mapillary-js/dist/mapillary.min.css";
 import EmbeddedMapModel from "./EmbeddedMapModel";
+import "./EmbeddedMap.css";
 
 export default function EmbeddedMap(
     props: LayoutElementProperties<EmbeddedMapModel>
 ): React.ReactElement {
     const { model } = props;
     const mlyRootEl = useRef<HTMLDivElement>();
+    const [sync, setSync] = useState(model.syncGcxMap);
```

Comment:
The purpose of `model.syncGcxMap` isn't quite clear to me. Should it instead be observable in the model? It seems like we have two sources of truth.

Reply:
Yes. I've changed this to use 'useWatchAndRerender' from the web api instead.